returnn/datasets/basic.py

```diff
@@ -1261,6 +1261,7 @@ def init_dataset(kwargs, extra_kwargs=None, default_kwargs=None):
   """
   assert kwargs
   if isinstance(kwargs, Dataset):
+    kwargs.initialize()
```

Comment:
I think, this is required for search to work properly. But I think someone with better knowledge of returnn should check. 
@albertz @patrick-wilken @dthulke  @JackTemaki

Reply:
Why is this only an issue for search? Afaik initialize isn't called somewhere else, so this should always cause an issue if the dataset passed to this method is not already initialised.

Reply:
It only crashed in search. But I do not know exactly, that is why I am asking.

Reply:
What is the crash/error?

Reply:
But also, I think this is probably just consistent to call this here. I think it also depends on the `Dataset` class. For many classes, this might also work without.

Reply:
The datasets defined in the config via `train`, `dev` and `eval` are initialized globally in `__main__().load_data()`. The search dataset set via `search_data` is not. That's the difference. So yes, this is needed here. (Assuming we don't want to require the dataset in the config to already be initialized.) However, for consistency we should initialize all datasets with this line here, also "train" etc. See my other comment...